src/main/java/seedu/rc4hdb/model/resident/fields/MatricNumber.java

```diff
@@ -1,36 +1,31 @@
 package seedu.rc4hdb.model.resident.fields;
 
-import static java.util.Objects.requireNonNull;
 import static seedu.rc4hdb.commons.util.AppUtil.checkArgument;
 
-import seedu.rc4hdb.commons.util.StringUtil;
-
 /**
  * Represents a Resident's matriculation number in RC4HDB.
  * Guarantees: immutable; is valid as declared in {@link #isValidMatricNumber(String)}
  */
-public class MatricNumber extends Field {
+public class MatricNumber extends ResidentField {
+
+    public static final String IDENTIFIER = "Matric";
```

Comment:
As mentioned in the other comment, the `IDENTIFIER` can be changed, if you want the code in `ResidentTableView.java` to be more uniform.

Reply:
I'm not too sure about the implications of changing the `IDENTIFIER`. But I doubt that there would be huge repercussions.